pkg/apis/azuredisk/v1beta2/types.go

```diff
@@ -521,3 +521,84 @@ type ListSnapshotsResult struct {
 	// +optional
 	NextToken string `json:"next_token,omitempty"`
 }
+
+type AzDiskDriverConfiguration struct {
+	Controller ControllerConfiguration `json:"controller"`
+	Node NodeConfiguration `json:"node"`
+	// The namespace where driver related custom resources are created
+	ObjectNamespace string `json:"objectNamespace"`
+	// CSI endpoint
+	Endpoint string `json:"endpoint"`
+	// node id
+	NodeID string `json:"nodeID"`
+	// Print the version and exit
+	Version bool `json:"version"`
+	// export the metrics
+	MetricsAddress string `json:"metricAddress"`
+	// Absolute path to the kubeconfig file. Required only when running out of cluster
+	Kubeconfig string `json:"kubeconfig"`
+	// name of the driver
+	DriverName string `json:"driverName"`
+	// boolean field to enable disk perf optimization
+	EnablePerfOptimization bool `json:"enablePerfOptimization"`
+	// cloud config secret name
+	CloudConfigSecretName string `json:"cloudConfigSecretName"`
+	// cloud config secret namespace
+	CloudConfigSecretNamespace string `json:"cloudConfigSecretNamesapce"`
+	// custom userAgent
+	CustomUserAgent string `json:"customUserAgent"`
+	// userAgent suffix
+	UserAgentSuffix string `json:"userAgentSuffix"`
+	// boolean field to enable csi-proxy GA interface on Windows
+	UseCSIProxyGAInterface bool `json:"useCSIProxyGAInterface"`
+	// boolean field to enable disk online resize
+	EnableDiskOnlineResize bool `json:"enableDiskOnlineResize"`
+	// Whether allow running driver without cloud config
+	AllowEmptyCloudConfig bool `json:"allowEmptyCloudConfig"`
+	// boolean field to enable async attach
+	EnableAsyncAttach bool `json:"enableAsyncAttach"`
+	// boolean field to enable ListVolumes on controller
+	EnableListVolumes bool `json:"enableListVolumes"`
+	// boolean field to enable ListSnapshots on controller
+	EnableListSnapshots bool `json:"enableListSnapshots"`
+	// boolean field to enable volume capacity check in CreateVolume
+	EnableDiskCapacityCheck bool `json:"enableDiskCapacityCheck"`
+	// QPS for the rest client. Defaults to 15
+	KubeClientQPS int `json:"kubeClientQPS"`
+	// vmss cache TTL in seconds (600 by default)
+	VmssCacheTTLInSeconds int64 `json:"vmssCacheTTLInSeconds"`
+}
```

Comment:
`CloudConfigSecretName`, `CloudConfigSecretNamespace`, `CustomUserAgent`, `UserAgentSuffix`, `AllowEmptyCloudConfig` and `VmssCacheTTLInSeconds` can all be grouped in a `CloudConfiguration` struct.

`KubeConfig`, and `KubeClientQPS` can be grouped in a `ClientConfiguration` struct.

`EnableDiskOnlineResize`, `EnableAsyncAttach`, `EnableListVolumes`, `EnableListSnapshots` and `EnableCapacityCheck` can be moved to ControllerConfiguration`.

`NodeID`, `EnablePerfOptimization` and `UseCSIProxyGAInterface` can be moved to `NodeConfiguration`.